src/components/avatar/avatar.ts

```diff
@@ -0,0 +1,76 @@
+import { html, LitElement } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { Constructor } from '../common/mixins/constructor.js';
+import { SizableMixin } from '../common/mixins/sizable.js';
+import { styles } from './avatar.material.css';
+
+export class IgcAvatarComponent extends SizableMixin(
+  <Constructor<LitElement>>LitElement
+) {
+  static styles = [styles];
+
+  @property()
+  src?: string;
+
+  @state()
+  private hasError = false;
+
+  @property()
+  alt?: string;
+
+  @property()
+  initials?: string;
+
+  @property({ reflect: true })
+  shape: 'circle' | 'rounded' | 'square' = 'square';
+
+  protected get classes() {
+    const { size, shape } = this;
+
+    return {
+      circle: shape === 'circle',
+      rounded: shape === 'rounded',
+      square: shape === 'square',
+      small: size === 'small',
+      medium: size === 'medium',
+      large: size === 'large',
+    };
+  }
+
+  constructor() {
+    super();
+    this.size = 'small';
+  }
+
+  render() {
+    return html`
+      <div
+        part="native"
```

Comment:
This part should be called `base` according to the spec.

Reply:
I would actually change the spec to be in line with all other components. The button and the badge now use `native` and the avatar shouldn't be the odd on in the bunch.

Reply:
If we want to be consistent across all components I think we should use `base` instead of `native` since it describes better that this is the base container element.

Reply:
True, fine then. Let's merge it like it is now, then we'll replace them all in a single PR.